app/javascript/components/pages/ApartmentShow.js

```diff
@@ -1,7 +1,56 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import { useParams } from 'react-router-dom'
+import { Container, Card, Button } from 'react-bootstrap'
 
-export default function ApartmentShow() {
+export default function ApartmentShow(props) {
+  let [apartment, setApartment] =useState([])
+  let { id } = useParams()
+
+  let {
+    logged_in,
+    current_user
+  } = props
+
+  console.log("show:", logged_in)
+  function showApartment (id) {
+    fetch(`/apartments/${id}`)
+    .then(request => request.json())
+    .then(payload => setApartment(payload))
+    .catch(err => console.log(err))
+  }
+  useEffect(() => {
+    showApartment(id)
+  },[])
```

Comment:
This works! Remember the project may call for utilizing props instead of redundant requests, something to mention in an interview about react.

Reply:
Will keep that in mind. I was wondering why my props weren't working when i passed it down as apartments but worked when i used apartment. I would like to pick your brain about this later.